### PR TITLE
Migrate to v0.14 API

### PR DESCRIPTION
**Pull request by cosmo0920**
When migrating v0.14 API, Fluentd permits to use EventTime and Integer in time part in event and it can contain nanosecond precision time.

### CODE REVIEW

**Comment by cosmo0920** [comment]
I've noticed that we can use built-in `timer` plugin helper in this `in_systemd` plugin class.

File: lib/fluent/plugin/in_systemd.rb [hunk 1]
```diff
@@ -76,12 +75,9 @@ def formatted(entry)
       end
 
       def watch
-        while @running
-          next unless @journal.wait(1_000_000)
```
Comment (errm) [inline lib/fluent/plugin/in_systemd.rb]: We should check that loosing this line does not effect performance, journal.wait returns straightaway if there are new events on the journal, here I think we are going to wait a full second once we have reached the end...

**Review by errm — COMMENTED** [review]
Not sure about this one ... everything else is looking great though... Thanks :)

File: lib/fluent/plugin/in_systemd.rb (hunk 1 above)
Comment (errm) [reply lib/fluent/plugin/in_systemd.rb]: I think this is fine ... we can always back out on this if anyone has specific issues.

**Comment by cosmo0920** [comment]
Thanks!

**Pull request merged**